File: src/main/java/xyz/lncvrt/galaxyboxpvp/events/EntityPickupItemListener.java
package xyz.lncvrt.galaxyboxpvp.events;

import xyz.lncvrt.galaxyboxpvp.GalaxyBoxPvP;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.EventPriority;
import org.bukkit.event.Listener;
import org.bukkit.event.entity.EntityPickupItemEvent;

public class EntityPickupItemListener implements Listener {
    private final GalaxyBoxPvP plugin;

    public EntityPickupItemListener(GalaxyBoxPvP plugin) {
        this.plugin = plugin;
    }

    @EventHandler(priority = EventPriority.LOWEST)
    public void onItemPickup(EntityPickupItemEvent event) {
        if (event.getEntity() instanceof Player player) plugin.convertInventoryItemsPrep(player, player.getUniqueId());
    }
}

File: src/main/kotlin/xyz/lncvrt/galaxyboxpvp/events/EntityPickupItemListener.kt
package xyz.lncvrt.galaxyboxpvp.events

import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.EventPriority
import org.bukkit.event.Listener
import org.bukkit.event.entity.EntityPickupItemEvent
import xyz.lncvrt.galaxyboxpvp.GalaxyBoxPvPPlugin

class EntityPickupItemListener(private val plugin: GalaxyBoxPvPPlugin) : Listener {
    @EventHandler(priority = EventPriority.LOWEST)
    fun onItemPickup(event: EntityPickupItemEvent) {
        if (event.getEntity() is Player) {
            val player = event.entity as Player
            plugin.convertInventoryItemsPrep(player, player.uniqueId)
        }
    }
}